packages/plugins/src/plugins/silo/siloRewardFetcher.ts

```diff
@@ -0,0 +1,110 @@
+import { EvmNetworkIdType } from '@sonarwatch/portfolio-core';
+import { Address, getAddress } from 'viem';
+import { logger } from '@nx/devkit';
+import { Fetcher, FetcherExecutor } from '../../Fetcher';
+import { getEvmClient } from '../../utils/clients';
+import {
+  PLATFORM_ID,
+  SILOS_INCENTIVE_CONTROLLER_ADDRESSES,
+  SILOS_REWARD_VAULTS_KEY,
+} from './constants';
+import { rewardAbi, rewardTokenAbi } from './abis';
+import { Cache } from '../../Cache';
+import { ElementRegistry } from '../../utils/elementbuilder/ElementRegistry';
+
+function fetcher(networkId: EvmNetworkIdType): Fetcher {
+  const executor: FetcherExecutor = async (owner: string, cache: Cache) => {
+    // Get reward vault addresses from cache
+    const rewardVaultAddresses = await cache.getItem<Address[]>(
+      SILOS_REWARD_VAULTS_KEY,
+      {
+        prefix: PLATFORM_ID,
+        networkId,
+      }
+    );
+
+    if (!rewardVaultAddresses) {
```

Comment:
empty arrays are truthy
```suggestion
    if (!rewardVaultAddresses || rewardVaultAddresses.length === 0) {
```